core/src/main/java/com/graphhopper/reader/osm/OSMNodeData.java

```diff
@@ -248,7 +250,7 @@ public void addCoordinatesToPointList(int id, PointList pointList) {
     }
 
     public void setTags(ReaderNode node) {
-        int tagIndex = nodeTagIndicesByOsmNodeIds.get(node.getId());
+        int tagIndex = (int) nodeTagIndicesByOsmNodeIds.get(node.getId());
```

Comment:
Maybe use `Math.toIntExact` so we get an error if the tag index is outside the int range?